src/azul/plugins/__init__.py

```diff
@@ -443,23 +442,31 @@ def list_sources(self,
         """
         raise NotImplementedError
 
-    @cached_property
-    def _source_ref_cls(self) -> Type[SOURCE_REF]:
-        cls = type(self)
-        base_cls = one(getattr(cls, '__orig_bases__'))
-        spec_cls, ref_cls = get_args(base_cls)
-        require(issubclass(spec_cls, SourceSpec))
-        require(issubclass(ref_cls, SourceRef))
+    @classmethod
+    @cache
+    def _get_params(cls) -> tuple:
```

Comment:
There aren't that many instances of concrete plugins so I doubt this class-based cache is worth it. Just compare the implementation of `@cache` with that of `@cached_property`. The latter incurs no state overhead on the instance. The former adds a lock and a dict. Considering that what we're caching here is cheap to begin with, the class cache just seems to much. 

If the method stays in the form of a cached property: the name is a little generic.